client/app/bundles/course/assessment/submission/selectors/history.ts

```diff
@@ -1,20 +1,53 @@
 import { AppState } from 'store';
+import { QuestionType } from 'types/course/assessment/question';
+import { SubmissionQuestionData } from 'types/course/assessment/submission/question/types';
+import {
+  AllAnswerItem,
+  CommentItem,
+} from 'types/course/assessment/submission/submission-question';
 
-import { AnswerHistory, QuestionHistory } from '../reducers/history/types';
-import { HistoryState } from '../types';
+import {
+  AnswerDetailsState,
+  FetchDataRequestStatus,
+  SubmissionHistoryState,
+} from '../reducers/history';
 
-const getLocalState = (state: AppState): HistoryState => {
+const getLocalState = (state: AppState): SubmissionHistoryState => {
   return state.assessments.submission.history;
 };
 
-export const getHistoryAnswers = (
-  state: AppState,
-): Record<number, AnswerHistory> => {
-  return getLocalState(state).answers;
-};
-
-export const getHistoryQuestions = (
-  state: AppState,
-): Record<number, QuestionHistory> => {
-  return getLocalState(state).questions;
-};
+export const getSubmissionQuestionHistory =
+  (submissionId: number, questionId: number) =>
+  (
+    state: AppState,
+  ): {
+    allAnswers: AllAnswerItem[];
+    answerDataById: Record<
+      number,
+      AnswerDetailsState<keyof typeof QuestionType>
+    >;
+    selectedAnswerIds: number[];
+    question:
+      | SubmissionQuestionData<keyof typeof QuestionType>
+      | null
+      | undefined;
+    canViewHistory: boolean;
+    comments: CommentItem[];
+    status: FetchDataRequestStatus | undefined;
+  } => {
+    const history = getLocalState(state);
+    return {
+      status: history[submissionId]?.[questionId]?.status,
+      allAnswers:
+        history[submissionId]?.[questionId]?.details?.allAnswers ?? [],
+      answerDataById:
+        history[submissionId]?.[questionId]?.details?.answerDataById ?? {},
+      selectedAnswerIds:
+        history[submissionId]?.[questionId]?.details
+          ?.sequenceViewSelectedAnswerIds ?? [],
+      question: history[submissionId]?.[questionId]?.details?.question,
+      canViewHistory:
+        history[submissionId]?.[questionId]?.details?.canViewHistory ?? false,
```

Comment:
There's a problem with this part, when this is a fresh submission, `canViewHistory` is `false` because `.[questionId]` and `.details` is not yet populated.

This means the [Past Answers (0)] chip is not being rendered.

To discuss: How about just leaving that chip as [Past Answers] without any count information, and only fetching on click? This should save us some querying and reducer logic as well.  
Generally, number of past answers is not critical, and users will only click [Past Answers] in the uncommon case where they wish to look back through historical answers.